src/OpenLoco/Scenario.cpp

```diff
@@ -50,10 +54,89 @@ namespace OpenLoco::Scenario
         call(0x004C4BC0);
     }
 
+    Season nextSeason(Season season)
+    {
+        switch (season)
+        {
+            case Season::autumn: return Season::winter;
+            case Season::winter: return Season::spring;
+            case Season::spring: return Season::summer;
+            case Season::summer: return Season::autumn;
+        }
```

Comment:
```
../src/OpenLoco/Scenario.cpp: In function 'OpenLoco::Scenario::Season OpenLoco::Scenario::nextSeason(OpenLoco::Scenario::Season)':
../src/OpenLoco/Scenario.cpp:66:5: error: control reaches end of non-void function [-Werror=return-type]
     }
     ^
```
This is a frustrating error, considering the limited values. What would be a sensible default? Just go with winter?

Reply:
I suppose we could also implement it as `(season + 1) & 3`, like the original game. I just thought this improved readability at little cost.